js/controllers.js

```diff
@@ -1,12 +1,192 @@
-var chartControllers = angular.module('chartControllers', []);
-
-chartControllers.controller('contactCtrl', ['$scope', '$http',
-  function ($scope, $http) {    
-      $scope.pageName = "contact";
-
-  }]);
-
-chartControllers.controller('chartCtrl', ['$scope', '$routeParams',
-  function($scope, $routeParams) {
-    $scope.pageName = "chartControllers";
-  }]);
\ No newline at end of file
+var chartControllers = angular.module('chartControllers', ['googlechart', 'nvd3', 'highcharts-ng']);
+
+
+chartControllers.controller('chartCtrl', ['$scope', '$location', 'Datax',
+    function($scope, $location, Datax) {
+        
+
+
+
+        //csv file data is load for all charts libraries
+        $scope.loadData = function(fileName) {   
+            //console.log("loading data "+fileName);
+            data = getData(fileName);
+            $scope.setActive('b1',fileName);
+            //$scope.Datax = Datax;
+            console.log(Datax)
+            loadYAxisLabel(fileName);
+            $scope.a = true;
+            drawChart();
+        };
+
+                //default one 
+        var myChart = {};

```

Comment:
you should consider making it a local variable and passing it to `drawChart`. there is no reason to keep it hanging in the outer scope